snooty/parser.py

```diff
@@ -644,6 +650,99 @@ def handle_tabset(self, node: n.Directive) -> None:
             key=lambda x: tabid_list.index(cast(n.Directive, x).options["tabid"]),
         )
 
+    def handle_wayfinding(self, node: n.Directive) -> None:
+        expected_options = specparser.Spec.get().wayfinding["options"]
+        expected_options_dict = {option.id: option for option in expected_options}
+        expected_child_opt_name = "wayfinding-option"
+        expected_child_desc_name = "wayfinding-description"
+        expected_children_names = {expected_child_opt_name, expected_child_desc_name}
+        wayfinding_name = "wayfinding"
+
+        valid_opts: List[n.Node] = []
+        valid_desc: n.Node | None = None
+        used_ids: Set[str] = set()
+
+        # Validate children
+        for child in node.children:
+            child_line_start = child.start[0]
+
+            invalid_child = None
+            if not isinstance(child, n.Directive):
+                # Catches additional unwanted types like Paragraph
+                invalid_child = child.type
+            elif not child.name in expected_children_names:
+                invalid_child = child.name
+
+            if invalid_child:
+                self.diagnostics.append(
+                    InvalidChild(
+                        invalid_child,
+                        wayfinding_name,
+                        f"{expected_child_opt_name} or {expected_child_desc_name}",
+                        child_line_start,
+                    )
+                )
+                continue
```

Comment:
i guess its a matter of `do we catch all errors in one go and not have to go through different errors every time` vs `catch early and return early`. (since this is the parser, and not post process) i'm leaning towards the former, but with all the CI integration time costs, i'm fine with catching more errors in the go as well.